// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTPeerEnums";
option optimize_for = LITE_RUNTIME;

enum PeerObjectType {
    PEEROBJECT_ERROR = 0;
    PEEROBJECT_MESSAGE = 1;
    PEEROBJECT_REQUEST = 2;
    PEEROBJECT_RESPONSE = 3;
    PEEROBJECT_PAYMENT = 4;
    PEEROBJECT_CASH = 5;
}

enum PeerRequestType {
    PEERREQUEST_ERROR = 0;
    PEERREQUEST_BAILMENT = 1;
    PEERREQUEST_OUTBAILMENT = 2;
    PEERREQUEST_PENDINGBAILMENT = 3;
    PEERREQUEST_CONNECTIONINFO = 4;
    PEERREQUEST_STORESECRET = 5;
    PEERREQUEST_VERIFICATIONOFFER = 6;
    PEERREQUEST_FAUCET = 7;
}

enum ConnectionInfoType {
    CONNECTIONINFO_ERROR = 0;
    CONNECTIONINFO_BITCOIN = 1;
    CONNECTIONINFO_BTCRPC = 2;
    CONNECTIONINFO_BITMESSAGE = 3;
    CONNECTIONINFO_BITMESSAGERPC = 4;
    CONNECTIONINFO_SSH = 5;
    CONNECTIONINFO_CJDNS = 6;
}

enum SecretType {
    SECRETTYPE_ERROR = 0;
    SECRETTYPE_BIP39 = 1;
}
